src/components/Item/VisualizationItem/Visualization/Visualization.js

```diff
@@ -79,16 +81,20 @@ class Visualization extends React.Component {
                                 <LoadingMask />
                             </div>
                         )}
-                        <VisualizationPlugin
-                            d2={this.context.d2}
-                            visualization={this.memoizedGetFilteredVisualization(
-                                pluginProps.visualization,
-                                pluginProps.itemFilters
+                        <D2Shim d2Config={{}}>
```

Comment:
Does this return the same `d2` instance if the config passed is different?
Ie. here there is no `schemas` passed, so is this causing the all schemas request to fire?

Reply:
If i understand the code correctly, `thed2` is a singleton and if it has previously been instantiated, then the initialization code does not run. https://github.com/dhis2/app-runtime-adapter-d2/blob/master/src/useD2.js#L43
So this should be safe. @amcgee , confirm?

And actually, this branch requests the schemas only once, while in prod, they are requested twice.

Reply:
Do we need any schema at all?

Reply:
Yes, we need the schemas. We tried removing them in 2.35, and the result was that all schemas were fetched. Then we added back only some of them, but things like d2-ui components were broken without them all. (e.g. organisationUnits)

Reply:
Try this in the dashboard repo:  `git show 4e3c83ec`

Reply:
👌